### PR TITLE
Support CCTP and noble auto deposits on testnet

### PR DESCRIPTION
**Pull request by rosepuppy**
Support CCTP and noble auto deposits on testnet.

Dependant on https://github.com/dydxprotocol/v4-clients/pull/80 and https://github.com/dydxprotocol/v4-abacus/pull/160

---

*public/configs/env.json*
  * Add `nobleValidator` to endpoints, it is used to initialize NobleClient

## Views

*`src/App.tsx`
  * Remove `SquidProvider`, it is no longer used.

* `src/views/forms/AccountManagementForms/DepositForm.tsx`
  * adjust notifications and slippage based on cctp

* `src/views/forms/AccountManagementForms/DepositForm/DepositButtonAndReceipt.tsx`
  * cctp does not have `bridgeFee` so we have to support that calculation
  * show "< 1 minutes" if estimated time less than 60s

* `src/views/forms/AccountManagementForms/WithdrawForm.tsx`
  * Adjust and slippage based on cctp

* `src/views/forms/AccountManagementForms/WithdrawForm/WithdrawButtonAndReceipt.tsx`
  * cctp does not have `bridgeFee` so we have to support that calculation
  * show "< 1 minutes" if estimated time less than 60s

* `src/views/notifications/TransferStatusNotification/index.tsx`
  * Fix empty object error bug

## Functions

* `src/lib/abacus/dydxChainTransactions.ts`
  * Add support for TransactionType.SendNobleIBC
  * Add support for QueryType.GetNobleBalance
  * Initialize NobleClient when both NobleWallet and nobleValidator URL is available

*  `src/lib/abacus/index.ts`
  * Enable SquidV2 for deposits only when not on mainnet
  * add setter for noble wallet

* `src/lib/squid.ts`
  * Create `fetchSquidStatus` for fetching squid status via API rather than SDK
  * We no longer want to use the SDK because it does not support V2.

## Hooks

* `src/hooks/useAccounts.tsx`
  * Create a NobleWallet when mnemonic is changed or detected, and pass it to abacus

* `src/hooks/useLocalNotifications.tsx`
  * Fetch squid status via API instead of SDK
  * Fix bug where squid status come back with empty error object  

## Packages

* bump abacus and v4-client
 

---

Let me know if you need testnet tokens for testing

### CODE REVIEW

File: src/lib/abacus/dydxChainTransactions.ts [hunk 1]
```diff
@@ -325,6 +335,46 @@ class DydxChainTransactions implements AbacusDYDXChainTransactionsProtocol {
...
+    try {
+      const ibcMsg = {
+        typeUrl: params.msgTypeUrl, // '/ibc.applications.transfer.v1.MsgTransfer',
+        value: params.msg,
+      };
+      const fee = await this.nobleClient.simulateTransaction([ibcMsg]);
+  
+      // take out fee from amount before sweeping
+      const amount = parseInt(ibcMsg.value.token.amount, 10) -
+        Math.floor(parseInt(fee.amount[0].amount, 10) * 1.4);
```
Comment (aforaleka) [inline src/lib/abacus/dydxChainTransactions.ts:357]: nit: set the `1.4` as a constant so it's clearer what that's doing?

File: src/lib/abacus/dydxChainTransactions.ts [hunk 2]
```diff
@@ -325,6 +335,46 @@ class DydxChainTransactions implements AbacusDYDXChainTransactionsProtocol {
...
+  
+      if (amount <= 0) {
```
Comment (aforaleka) [inline src/lib/abacus/dydxChainTransactions.ts:359]: should `amount === 0` be ok?

File: src/lib/abacus/dydxChainTransactions.ts [hunk 3]
```diff
@@ -441,6 +496,11 @@ class DydxChainTransactions implements AbacusDYDXChainTransactionsProtocol {
           const parseDelegations = this.parseToPrimitives(delegations);
           callback(JSON.stringify(parseDelegations));
           break;
+        case QueryType.GetNobleBalance:
+          const nobleBalance = await this.nobleClient?.getAccountBalance('uusdc');
```
Comment (aforaleka) [inline src/lib/abacus/dydxChainTransactions.ts:500]: umm i thought `uusdc` is only used in gas -- does the ibc/.. denom also work?

Comment (rosepuppy) [reply src/lib/abacus/dydxChainTransactions.ts:500]: This is for the noble chain not our chain, which only uses `uusdc` for denom for testnet and mainnet. we have it hardcoded in v4-client and abacus as well.

File: src/lib/abacus/dydxChainTransactions.ts (hunk 2 above)
Comment (aforaleka) [reply src/lib/abacus/dydxChainTransactions.ts:359]: ah i meant should we only check for negative amount, like if (amount < 0)?

Comment (rosepuppy) [reply src/lib/abacus/dydxChainTransactions.ts:359]: ohhh gotcha, I can make it <= 0

**Review by aforaleka — APPROVED** [review]

**Pull request merged**